### DISCUSSION TITLE
How to make the "Animated Textures" work ???

### DISCUSSION THREAD

**Discussion by Cebolitoss** [google/model-viewer#4068] [Q&A]
I'm trying to run a sample where I run a video on a 3D model, but I kept getting errors. To solve this I tried downloading the whole "model-viewer-master" and run it locally.

After the whole "model-viewer-master" was in my local server I kept receiving errors as 404 (Not Found) for lots of files as the files inside the "built" folder, the files inside the "node_modules\focus-visible\dist" folder. So I tried to fix the folders structure inspecting the original (https://modelviewer.dev), copying the files from the website folders structure straight to my local server. And once I fixed all the 404 (Not Found) for the "built" and "node_modules" folder, I kept receiving the "modelViewerAnimated.createVideoTexture is not a function" error, and lots of others.

I used the following online "model-viewer.js" to fix the node_modules 404 errors as bellow, but kept receiving the errors:

<script type="module"
      src="https://unpkg.com/@google/model-viewer/dist/model-viewer.js"">
  </script>

I'm really lost on what I'm doing wrong.

This are the errors that I kept receiving after fixing everything that I thought that would help.

![image](https://user-images.githubusercontent.com/117696201/215039751-3a865b46-e1b0-4527-aabd-b420c63f24c4.png)

What should I do to fix it?

**Comment by Cebolitoss** [comment]
These are the more recent ones:

![image](https://user-images.githubusercontent.com/117696201/215044166-691f351b-4d08-4b73-8ce9-b748dad3c26b.png)

**Comment by Cebolitoss** [comment]
I feel like I'm nearly there, but also feel like I'm still far: 😭 😭😂

I'm not running the whole model-viewer-master anymore, is my actual folders structure.

![image](https://user-images.githubusercontent.com/117696201/215046807-39566f1a-a969-4213-b607-c111be57731f.png)

**Reply by elalish** [reply]
Best guess is you're not actually using the lastest pre-release MV yet, try following this FAQ: https://modelviewer.dev/docs/faq.html#entrydocs-general-questions-api

**Reply by planetMatrix** [reply]
@elalish I'm facing the exact same as

> I feel like I'm nearly there, but also feel like I'm still far: 😭 😭😂
> 
> I'm not running the whole model-viewer-master anymore, is my actual folders structure.
> 
> ![image](https://user-images.githubusercontent.com/117696201/215046807-39566f1a-a969-4213-b607-c111be57731f.png)

Could you please guide us which is the right version of js file that has createVideoTexture function.

Many thanks in Advance

**Reply by planetMatrix** [reply]
Interestingly the same is the issue with the official example
https://modelviewer.dev/examples/scenegraph/#animatedTexturesExample

**Reply by Cebolitoss** [reply]
Yes I'm having this issue with the example showed in the official example. So @elalish in the example (here: https://modelviewer.dev/examples/scenegraph/#animatedTexturesExample) it says that I should import "../../node_modules/@google/model-viewer/dist/model-viewer.min.js" but in the FAQ that you sent me says the opposite!  

https://modelviewer.dev/node_modules/@google/model-viewer/dist/model-viewer.min.js. Do not use this URL for production!

What should I do? Where can I get the right version of js file that has createVideoTexture function ???

Could you point me out please?

**Reply by planetMatrix** [reply]
@Cebolitoss use this version.... it works!
https://modelviewer.dev/node_modules/@google/model-viewer/dist/model-viewer.min.js

**Reply by Cebolitoss** [reply]
Thank for your reply @planetMatrix. I was able to run the video, but it doesn't have audio, did you have the same issue?
Other thing is, it opens on AR on android, but it doesn't open the AR on IOs.

On IOs it runs in the webpage, but doesn't open the AR.

Did you try it on AR or only in the webpage ??
Were you able to run on IOs too?

**Reply by elalish** [reply]
Okay, it sounds like you both got this working on the MV pre-release version? We'll do a release soon, but remember the pre-release is only for testing. And let me know if there's anything we can do to clarify the documentation.

The sound is muted by default; you can change all the playback settings by getting the underlying element: `videoTexture.source.element` https://modelviewer.dev/examples/scenegraph/#pickMaterialExample

Video textures are not supported in glTF, QuickLook, or SceneViewer, so the only AR experience this will work on is WebXR on Android, as is the case for most of our advanced features.

**Reply by ghost** [reply]
@elalish The video textures are still not supported for Quicklook or Sceneviewer? (For showing video in AR mode in iOS phones)

**Reply by elalish** [reply]
No, they're not, and I wouldn't expect that they will be any time soon, if ever. This is why we prefer WebXR.

**Reply by ghost** [reply]
@elalish Thank you for your quick reply. Do you have any idea how can I play a video in AR mode for both Android and iOS? It seems is not possible with model-viewer. right?

**Reply by elalish** [reply]
No, sadly our hands are tied on iOS.
